### PR TITLE
[nrfconnect] Added resolvable private address for BLE device

### LINKED ISSUE #5425: BLE address is not changing for the following commissioning attempts.
#### Problem
According to spec 5.2.3.8.5: To ensure privacy, the BLE controller SHALL use LE Random Device Address. An uncommissioned
CHIP device shall change the address on every successful or unsuccessful commissioning attempt or on an announcement timeout. That test case is also described in the Test Event 1 test plan TC#7 Step 6.

 #### Proposed Solution
Implement using random device BLE address and changing it on every commissioning attempt.

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
#### Problem
According to test event 1 test plan, CHIP device should use new address on each successfull or unsuccessfull commissioning
attempt.

 #### Summary of Changes
* Added configs enabling BT_PRIVACY
* Added generating BLE resolvable private address instead of static one.

 Fixes #5425 for nrfconnect

### CODE REVIEW

**Comment by markus-becker-tridonic-com** [comment]
add label TE1 to this PR.

**Review by saurabhst — APPROVED** [review]

File: src/platform/Zephyr/BLEManagerImpl.cpp [hunk 1]
```diff
@@ -202,14 +204,20 @@ CHIP_ERROR BLEManagerImpl::StartAdvertising(void)
...
     if (!mFlags.Has(Flags::kAdvertising))
     {
+// If necessary, inform the ThreadStackManager that CHIPoBLE advertising is about to start.
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
         ThreadStackMgr().OnCHIPoBLEAdvertisingStart();
-    }
 #endif // CHIP_DEVICE_CONFIG_ENABLE_THREAD
 
+        // Generate new private BLE address
+        struct bt_le_oob bleOobInfo;
```
Comment (Damian-Nordic) [inline src/platform/Zephyr/BLEManagerImpl.cpp:215]: nit: `struct` is unnecessary

**Review by Damian-Nordic — APPROVED** [review]

**Comment by tcarmelveilleux** [comment]
Why resolvable private address? Resolvable private addresses imply use of LE security and bonding to make sense of them. Why not use Static Random address like the spec specifies:

> To ensure privacy, the BLE controller SHALL use LE Random Device Address. A commissionable CHIP device shall change its Advertising Address on every successful or unsuccessful commissioning attempt or on an announcement timeout (see Announcement Duration).

**Comment by kkasperczyk-no** [comment]
> Why resolvable private address?
>
I felt like using private address is basically to hide device identity and prevent from tracking, so to ensure privacy. If it comes to the static random addresses I saw use cases describing it as rather good for assigning random address on boot, but not to change it at runtime multiple times. So basing on that and the fact that according to spec address should be random (it does not clearly say about private or static) I figured out that private will be more relevant.

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
NOTE: CP done but it was not clean (additional change on advertiser flags)

**Pull request merged**